### PR TITLE
Add `--random-seed` option to specify a fixed seed for better reproducibility

### PR DESCRIPTION
**Pull request by myaaaaaaaaa**
Streamlines the process of rolling for a good fixed seed when debugging games that have randomization:

https://github.com/godotengine/godot/issues/70406#issuecomment-1399112562
https://github.com/godotengine/godot/issues/73431#issuecomment-1435456566

Also has benefits for benchmarking:

https://github.com/godotengine/godot-benchmarks/blob/234bd8ecc0c7c59a12519cfc6d60d82efa8eaaac/main.gd#L14-L17

### CODE REVIEW

File: main/main.cpp [hunk 1]
```diff
@@ -768,6 +769,9 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 	packed_data->add_pack_source(zip_packed_data);
 #endif
 
+	// Initialize rng before argument parsing so that --seed can override it
+	Math::randomize();
```
Comment (Calinou) [inline main/main.cpp:773]: `randomize()` is already called implicitly on startup on 4.0, so I'd remove that other initialization to prevent it from being initialized twice.

**Comment by Calinou** [comment]
This CLI argument should probably be restricted to debug builds only, as it could be used for cheating by using a very favorable seed for the player.

File: main/main.cpp [hunk 2]
```diff
@@ -427,6 +427,7 @@ void Main::print_help(const char *p_binary) {
 #endif
 	OS::get_singleton()->print("  --frame-delay <ms>                Simulate high CPU load (delay each frame by <ms> milliseconds).\n");
 	OS::get_singleton()->print("  --time-scale <scale>              Force time scale (higher values are faster, 1.0 is normal speed).\n");
+	OS::get_singleton()->print("  --seed <seed>                     Specify a fixed seed for better reproducibility.\n");
```
Comment (Calinou) [inline main/main.cpp:430]: Only the global RandomNumberGenerator instance is affected by this CLI argument, not local instances that may be created in game logic:

```suggestion
	OS::get_singleton()->print("  --seed <seed>                     Specify a fixed random number generation seed for the global RandomNumberGenerator instance for better reproducibility.\n");
```

File: main/main.cpp (hunk 1 above)
Comment (myaaaaaaaaa) [reply main/main.cpp:773]: Are you referring to the removed line in `SceneTree::SceneTree()`? I tried commenting this line out, but it resulted in a fixed seed, so there doesn't appear to be anywhere else where the rng is initialized.

Comment (Calinou) [reply main/main.cpp:773]: I didn't notice you removed the other `randomize()` call, so this should be good to go.

File: main/main.cpp (hunk 2 above)
Comment (myaaaaaaaaa) [reply main/main.cpp:430]: > Only the global RandomNumberGenerator instance is affected by this CLI argument

I am currently investigating ways to change `randomize()` to transparently [split its new seed from the global instance](https://jax.readthedocs.io/en/latest/jax.random.html#prng-keys)

**Comment by myaaaaaaaaa** [comment]
> This CLI argument should probably be restricted to debug builds only, as it could be used for cheating by using a very favorable seed for the player (at least in the early stages of gameplay). See also #33965.

Done

> Also, if [godotengine/godot-proposals#6226](https://github.com/godotengine/godot-proposals/issues/6226) is implemented, we should probably make it so that this CLI argument also sets the initial particle seed for all particles.

> Only the global RandomNumberGenerator instance is affected by this CLI argument

I am currently investigating ways to change `randomize()` to transparently [split its new seed from the global instance](https://jax.readthedocs.io/en/latest/jax.random.html#prng-keys), which should handle nearly any usage of random number generation (barring cases like the user manually seeding with the time, or creating local `RandomNumberGenerator`s in threads)
